firewood/src/merkle/node.rs

```diff
@@ -934,6 +933,52 @@ mod tests {
         check_node_encoding(node);
     }
 
+    #[test_case(&[])]
+    #[test_case(&[0x00])]
+    #[test_case(&[0x0F])]
+    #[test_case(&[0x00, 0x00])]
+    #[test_case(&[0x01, 0x02])]
+    #[test_case(&[0x00,0x0F])]
+    #[test_case(&[0x0F,0x0F])]
+    #[test_case(&[0x0F,0x01,0x0F])]
+
+    fn encoded_node_branch_node_bincode_serialize(path_nibbles: &[u8]) -> Result<(), Error> {
+        let node: EncodedNode<Bincode> = EncodedNode::new(EncodedNodeType::Branch {
+            path: PartialPath(path_nibbles.to_vec()),
+            children: Default::default(),
+            value: Some(Data(vec![1, 2, 3, 4])),
+        });
+
+        let bincode = Bincode::new();
+
+        let node_bytes = bincode.serialize_impl(&node)?;
```

Comment:
Seems as though the docs are missing, but this isn't how you're supposed to use this. The `_impl` methods are only meant to be called by the default `BinarySerde` trait methods. 
```suggestion
        let node_bytes = Bincode::serialize(&node)?;
```